src/index.ts

```diff
@@ -0,0 +1,134 @@
+import escape from 'xml-escape';
+import * as Cypress from 'cypress';
+import BeforeRunDetails = Cypress.BeforeRunDetails;
+import PluginConfigOptions = Cypress.PluginConfigOptions;
+import PluginEvents = Cypress.PluginEvents;
+import Spec = Cypress.Spec;
+import { ConfigOption, JUnitTestSuite } from './type';
+import Reporter, { TestCase, TestSuite } from './reporter';
+
+let reporter: Reporter;
+
+const onBeforeRun = (details: BeforeRunDetails) => {
```

Comment:
Style: May I suggest we use named functions across this project? I know that we have some inconsistencies across our JS/TS projects, but when creating something new, we should stick to one standard 🙏 This would apply to all of the anonymous functions declared in this project.

```suggestion
function onBeforeRun(details: BeforeRunDetails) {
```